common/doordeck.js

```diff
@@ -1,51 +1,61 @@
-import { AUTH_TOKEN } from './constants'
+import {AUTH_TOKEN, BASE_URL, DEFAULT_BASE_URL} from './constants'
 import ephemaralKeyGenerator from './services/ephemeralKeyGenerator'
 import certificate from './services/certificate'
 import libSodium from 'libsodium-wrappers'
 import device from "./services/deviceOperation";
 
 const isLoaded = async function (authToken) {
-  if (authToken === getStoredAuthToken()) {
-    if (ephemaralKeyGenerator.retrieveSavedKeys() !== null) {
-      return (await certificate.retrieveSavedCert()) !== null;
+    if (authToken === getStoredAuthToken()) {
+        if (ephemaralKeyGenerator.retrieveSavedKeys() !== null) {
+            return (await certificate.retrieveSavedCert()) !== null;
+        } else return false
     } else return false
-  } else return false
-}
-
-const doordeckInit = function (authToken) {
-  return new Promise (function(resolve, reject) {
-    libSodium.ready.then(async function () {
-      if ((await isLoaded())) resolve({state: 'success', message: 'Doordeck is already initialised.'})
-      if (authToken !== null && authToken !== undefined) {
-        storeAuthToken(authToken)
-        ephemaralKeyGenerator.generateKeys().then(keys => {
-          certificate.getCertificate(keys).then(response => {
-            resolve(response)
-          }, fail => {
-            reject(fail)
-          })
+}
+
+const doordeckInit = function (authToken, baseUrl) {
+    return new Promise(function (resolve, reject) {
+        libSodium.ready.then(async function () {
+            if ((await isLoaded())) resolve({state: 'success', message: 'Doordeck is already initialised.'})
+            if (authToken !== null && authToken !== undefined) {
+                storeBaseUrl(baseUrl)
+                storeAuthToken(authToken)
+                ephemaralKeyGenerator.generateKeys().then(keys => {
+                    certificate.getCertificate(keys).then(response => {
+                        resolve(response)
+                    }, fail => {
+                        reject(fail)
+                    })
+                })
+            } else reject({state: 'error', message: 'No Auth Token provided.'})
         })
-      } else reject({state: 'error', message: 'No Auth Token provided.'})
     })
-  })
 }
 
 const storeAuthToken = function (authToken) {
-  localStorage[AUTH_TOKEN] = authToken
+    localStorage[AUTH_TOKEN] = authToken
+}
+
+const storeBaseUrl = function (baseUrl) {
+    localStorage[BASE_URL] = baseUrl
+}
+
+const getBaseUrl = function () {
```

Comment:
if we default the initialiser as mentioned, we wouldn't need this I believe?

Reply:
I kept the `default_base_url` if the baseUrl is set to null or empty from init. 
- Or we can force an error if it is not set.
- Instead of asking `baseUrl` we can ask for environment e.g. `Production`, `Staging`

Reply:
> Instead of asking baseUrl we can ask for environment e.g. Production, Staging
no, it has to be URL/environment agnostic. We'd provide a base Url at the beginning which we can default to the production one